calcite-adapter/src/test/java/software/amazon/documentdb/jdbc/metadata/DocumentDbTableSchemaGeneratorTest.java

```diff
@@ -1077,15 +1077,15 @@ void testInconsistentArrayDataType() {
         final Map<String, String> tableNameMap = new HashMap<>();
         final BsonDocument[] tests = new BsonDocument[] {
                 BsonDocument.parse(
-                        "{ \"_id\" : \"key\", \"array\" : [ 1, [ 2 ] ] }"),
+                        "{ \"_id\" : \"key\", \"array\" : [ 1, [ 2 ], null ] }"),
                 BsonDocument.parse(
                         "{ \"_id\" : \"key\", \"array\" : [ 1, { \"field\" : 2 } ] }"),
                 BsonDocument.parse(
                         "{ \"_id\" : \"key\", \"array\" : [ { \"field\" : 2 }, 3 ] }"),
                 BsonDocument.parse(
                         "{ \"_id\" : \"key\", \"array\" : [ { \"field\" : 2 }, [ 3, 4 ] ] }"),
                 BsonDocument.parse(
-                        "{ \"_id\" : \"key\", \"array\" : [ [ 1, 2 ], { \"field\" : 2 } ] }"),
+                        "{ \"_id\" : \"key\", \"array\" : [ [ 1, 2 ], { \"field\" : 2 }, null ] }"),
```

Comment:
Do we have a test where the field value might be null or another array with a null element?

Reply:
Not for nested arrays. Will add! 👍